cmd/deployment/template/show.go

```diff
@@ -0,0 +1,54 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package cmddeploymenttemplate
+
+import (
+	"github.com/elastic/cloud-sdk-go/pkg/api/deploymentapi/deptemplateapi"
+	"github.com/spf13/cobra"
+
+	"github.com/elastic/ecctl/pkg/ecctl"
+)
+
+var showCmd = &cobra.Command{
+	Use:     "show",
+	Short:   "Displays a deployment template",
+	PreRunE: cobra.ExactArgs(1),
+	RunE: func(cmd *cobra.Command, args []string) error {
+		stackVersion, _ := cmd.Flags().GetString("stack-version")
+		hideIC, _ := cmd.Flags().GetBool("hide-instance-configurations")
+
+		res, err := deptemplateapi.Get(deptemplateapi.GetParams{
+			API:                        ecctl.Get().API,
+			Region:                     ecctl.Get().Config.Region,
+			StackVersion:               stackVersion,
+			HideInstanceConfigurations: hideIC,
+			TemplateID:                 args[0],
+		})
+		if err != nil {
+			return err
+		}
+
+		return ecctl.Get().Formatter.Format("", res)
+	},
+}
+
+func init() {
+	Command.AddCommand(showCmd)
+	showCmd.Flags().Bool("hide-instance-configurations", false, "Hides instance configurations - only visible when using the JSON output")
+	showCmd.Flags().String("stack-version", "", "It will cause the returned deployment templates which are valid for the specified stacks version.")
```

Comment:
is this flag necessary if you're only returning the specified template?

```suggestion
	showCmd.Flags().String("stack-version", "", "Optional filter to only return deployment templates which are valid for the specified stack version.")
```

Reply:
Apparently it's a valid argument, so I left it because it's available on the API. I guess it might be useful if you're trying to verify that a DT is compatible with stack version X.Y.Z

Reply:
true 🤔